gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -51,6 +55,53 @@ case class StringTranslateTransformer(substraitExprName: String, srcExpr: Expres
   }
 }
 
+case class StringLocateTransformer(
+  substraitExprName: String,
+  substrExpr: ExpressionTransformer,
+  strExpr: ExpressionTransformer,
+  startExpr: ExpressionTransformer,
+  original: StringLocate) extends ExpressionTransformer with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(
+      GlutenConfig.GLUTEN_CLICKHOUSE_BACKEND)) {
+
+      val substrNode = substrExpr.doTransform(args)
+      val strNode = strExpr.doTransform(args)
+      val startNode = startExpr.doTransform(args)
+
+      // Special Case
+      // In Spark, return 0 when start_pos is null
+      // but when start_pos is not null, return null if either str or substr is null
+      // so we need convert it to if(isnull(start_pos), 0, position(substr, str, start_pos)
```

Comment:
if start_pos is 0, CH backend return 0 or null?

Reply:
If start_pos is 0 and both str and substr are not null, return 0. 
However, if start_pos is 0 and either str or substr is null,  return null, which is consistent with the behavior of Spark.